shell/platform/common/cpp/text_input_model.h

```diff
@@ -111,15 +160,39 @@ class TextInputModel {
   // The position of the cursor.
   int selection_extent() const { return selection_.extent(); }
 
+  // The position where the composing region starts.
+  int composing_base() const { return composing_range_.base(); }
+
+  // The position where the composing region ends.
+  int composing_extent() const { return composing_range_.extent(); }
+
+  bool composing() const { return composing_; }
+
  private:
   // Deletes the current selection, if any.
   //
   // Returns true if any text is deleted. The selection base and extent are
   // reset to the start of the selected range.
   bool DeleteSelected();
 
+  // Adjusts the composing range to |length|.
+  void SetComposingLength(size_t length);
+
+  // Returns the currently editable text range.
+  //
+  // In composing mode, returns the composing range; otherwise, returns a range
+  // covering the entire text.
+  TextRange editing_range() const {
```

Comment:
Nit: s/editing/editable/ ? It sounded like it might be the selected area from the current name when I was reading implementation code using it.

Reply:
I like it; also avoids it potentially sounding like a bool. Done.